nixos/modules/services/networking/create_ap.nix

```diff
@@ -0,0 +1,159 @@
+    enable = mkEnableOption "Enable Create Access Point Service";
+
+    extraConfig = mkOption {
+      description = ''
+        Extra config (see <link xlink:href="https://github.com/oblique/create_ap/blob/master/create_ap.conf"/>).
+      '';
+      type = lines;
+      example = ''
+        CHANNEL=default
+        GATEWAY=10.0.0.1
+        WPA_VERSION=2
+        ETC_HOSTS=0
+        DHCP_DNS=gateway
+        NO_DNS=0
+        NO_DNSMASQ=0
+        HIDDEN=0
+        MAC_FILTER=0
+        MAC_FILTER_ACCEPT=/etc/hostapd/hostapd.accept
+        ISOLATE_CLIENTS=0
+        SHARE_METHOD=nat
+        IEEE80211N=0
+        IEEE80211AC=0
+        HT_CAPAB=[HT40+]
+        VHT_CAPAB=
+        DRIVER=nl80211
+        NO_VIRT=0
+        COUNTRY=
+        FREQ_BAND=2.4
+        NEW_MACADDR=
+        DAEMONIZE=0
+        NO_HAVEGED=0
+        WIFI_IFACE=wlan0
+        INTERNET_IFACE=eth0
+        SSID=MyAccessPoint
+        PASSPHRASE=12345678
+        USE_PSK=0
+      '';
+      default = "";
+    };
+
+    wifi-iface = mkOption {
+      description = ''
+        Wi-Fi interface to use (Use `ip link show` to list available).
+      '';
+      type = str;
+      example = "wlan0";
+    };
+
+    internet-iface = mkOption {
+      description = ''
+        Interface to use for internet connection (Use `ip link show` to list available).
+      '';
+      type = str;
+      example = "enp0";
+    };
+
+    ssid = mkOption {
+      description = "SSID of the access point.";
+      type = str;
+      example = "MyAccessPoint";
+    };
+
+    passphrase = mkOption {
+      description = "Passphrase to use for access point.";
+      type = str;
+      example = "12345678";
+    };
+
+    gateway = mkOption {
+      description = "IPv4 Gateway for the Access Point.";
+      type = str;
+      example = "10.0.0.1";
+    };
+
+    channel = mkOption {
+      type = either str int;
+      example = 1;
+      default = "default";
+      description = "Channel number.";
+    };
+  };
+
+  ##### implementation
+  config = let capabilities =
+               [ "CAP_CHOWN"
+                 "CAP_DAC_READ_SEARCH"
+                 "CAP_KILL"
+                 "CAP_NET_ADMIN"
+                 "CAP_NET_RAW"
+                 "CAP_NET_BIND_SERVICE" ];
+  in mkIf cfg.enable {
+    environment.systemPackages =  [ pkgs.create_ap ];
+
+    systemd.services.create_ap = {
+      description = "Create AP Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target"
+                "sys-subsystem-net-devices-${cfg.wifi-iface}.device"
+                "sys-subsystem-net-devices-${cfg.internet-iface}.device" ];
+      bindsTo = [ "sys-subsystem-net-devices-${cfg.wifi-iface}.device"
+                  "sys-subsystem-net-devices-${cfg.internet-iface}.device" ];
+      serviceConfig = {
+        KillSignal = "SIGINT";
+        Restart = "on-failure";
+        RestartSec = 5;
+        DynamicUser = true;
+        ProtectSystem = true;
```

Comment:
`ProtectSystem` also supports the values `"full"` and `"strict"` which are even more limiting than `true` btw, not sure if they work for this